src/qtism/data/content/BodyElement.php

```diff
@@ -272,4 +329,290 @@ public function getDir()
+        return $this->ariaControls;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaControls()
+    {
+        return $this->ariaControls !== '';
+    }
+
+    /**
+     * @param string $ariaDescribedBy
+     * @throws InvalidArgumentException
+     */
+    public function setAriaDescribedBy($ariaDescribedBy)
+    {
+        if ($ariaDescribedBy !== '' && !Format::isAriaIdRefs($ariaDescribedBy)) {
+            $val = (is_object($ariaDescribedBy)) ? ('instance of ' . get_class($ariaDescribedBy)) : $ariaDescribedBy;
+
+            $msg = "'${val}' is not a valid value for attribute 'aria-describedby'.";
+            throw new InvalidArgumentException($msg);
+        }
+
+        $this->ariaDescribedBy = $ariaDescribedBy;
+    }
+
+    /**
+     * @return string
+     */
+    public function getAriaDescribedBy()
+    {
+        return $this->ariaDescribedBy;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaDescribedBy()
+    {
+        return $this->ariaDescribedBy !== '';
+    }
+
+    /**
+     * @param string $ariaFlowTo
+     * @throws InvalidArgumentException
+     */
+    public function setAriaFlowTo($ariaFlowTo)
+    {
+        if ($ariaFlowTo !== '' && !Format::isAriaIdRefs($ariaFlowTo)) {
+            $val = (is_object($ariaFlowTo)) ? ('instance of ' . get_class($ariaFlowTo)) : $ariaFlowTo;
+
+            $msg = "'${val}' is not a valid value for attribute 'aria-flowto'.";
+            throw new InvalidArgumentException($msg);
+        }
+
+        $this->ariaFlowTo = $ariaFlowTo;
+    }
+
+    /**
+     * @return string
+     */
+    public function getAriaFlowTo()
+    {
+        return $this->ariaFlowTo;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaFlowTo()
+    {
+        return $this->ariaFlowTo !== '';
+    }
+
+    /**
+     * @param string $ariaLabelledBy
+     * @throws InvalidArgumentException
+     */
+    public function setAriaLabelledBy($ariaLabelledBy)
+    {
+        if ($ariaLabelledBy !== '' && !Format::isAriaIdRefs($ariaLabelledBy)) {
+            $val = (is_object($ariaLabelledBy)) ? ('instance of ' . get_class($ariaLabelledBy)) : $ariaLabelledBy;
+
+            $msg = "'${val}' is not a valid value for attribute 'aria-labelledby'.";
+            throw new InvalidArgumentException($msg);
+        }
+
+        $this->ariaLabelledBy = $ariaLabelledBy;
+    }
+
+    /**
+     * @return string
+     */
+    public function getAriaLabelledBy()
+    {
+        return $this->ariaLabelledBy;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasAriaLabelledBy()
+    {
+        return $this->ariaLabelledBy !== '';
+    }
+
+    /**
+     * @param string $ariaOwns
+     * @throws InvalidArgumentException
+     */
+    public function setAriaOwns($ariaOwns)
+    {
+        if ($ariaOwns !== '' && !Format::isAriaIdRefs($ariaOwns)) {
+            $val = (is_object($ariaOwns)) ? ('instance of ' . get_class($ariaOwns)) : $ariaOwns;
```

Comment:
does it have any sense to use this validation? (is_object)

Reply:
Yes because it is possible that somebody gives it a try. We can solve this by strong typing the parameter, but I prefer we do this when we will tackle the `strict_type` milestone :smile: